enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/state/machines/id/ReplicatedIdGenerator.java

```diff
@@ -35,26 +39,31 @@
     private final Log log;
     private final ReplicatedIdRangeAcquirer acquirer;
     private volatile long highId;
-    private volatile long defragCount;
     private volatile IdRangeIterator idQueue = EMPTY_ID_RANGE_ITERATOR;
 
-    ReplicatedIdGenerator( IdType idType, long highId, ReplicatedIdRangeAcquirer acquirer, LogProvider
-            logProvider )
+    private IdContainer idContainer;
+
+    ReplicatedIdGenerator( FileSystemAbstraction fs, File file, IdType idType, long highId,
+            ReplicatedIdRangeAcquirer acquirer, LogProvider logProvider, int grabSize, boolean aggressiveReuse )
     {
         this.idType = idType;
         this.highId = highId;
         this.acquirer = acquirer;
         this.log = logProvider.getLog( getClass() );
+        idContainer = new IdContainer( fs, file, grabSize, aggressiveReuse );
+        idContainer.init();
     }
 
     @Override
     public void close()
     {
+        idContainer.close( highId );
     }
 
     @Override
     public void freeId( long id )
     {
+        idContainer.freeId( id );
```

Comment:
`IdGeneratorImpl` protects its `IdContainer` by making its methods `synchronized`. This class does not do that, so does this not expose the `IdContainer` to concurrency that it is not prepared to handle?